### PR TITLE
Support refreshing viewId in modals properly

### PR DESCRIPTION
**Pull request by siemiatj**
Related to #7019

### CODE REVIEW

**Comment by TheBestPessimist** [comment]
I took a look over the code, but this is waaaaay over my level. `¯\_(ツ)_/¯`

Do adding some tests make sense here?

**Comment by petrican** [comment]
I've done some testing and overall it looks good but if I am to use the filter I get again no data (this might be an issue on BE side as I see it received the filter) - pls se the video https://www.loom.com/share/e3d47807d676437e9b93ed180788e02b @siemiatj @TheBestPessimist

**Comment by petrican** [comment]
Also tests for the view actions should be updated/added if possible with this PR or with a separate one (pls open issue for that such that we won't forget). Thanks

**Comment by petrican** [comment]
I have run the e2e suite https://jenkins.metasfresh.com/job/ops/job/run_e2e_tests/947/ and all tests are passing . Tested against  https://dev1219.metasfresh.com   deployed with  5.146.2_3_frontend7019 build

File: frontend/src/actions/TableActions.js [hunk 1]
```diff
@@ -277,8 +273,12 @@ export function updateGridTable(tableId, tableResponse) {
...
         return Promise.resolve(true);
       } else {
-        const windowType = tableResponse.windowType || tableResponse.windowId;
-        const tableLayout = getView(getState(), windowType).layout;
+        const isModal = !!tableResponse.modalId;
+        const windowId = isModal
+          ? tableResponse.modalId
+          : tableResponse.windowType || tableResponse.windowId;
+
+        const tableLayout = getView(getState(), windowId, isModal).layout;
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply frontend/src/actions/TableActions.js:281]: No, as `getView` is a selector. And we don't use layout elsewhere so what for ? This is perfectly sufficient.

**Comment by siemiatj** [comment]
> Also tests for the view actions should be updated/added if possible with this PR or with a separate one (pls open issue for that such that we won't forget). Thanks

Yeah I'm aware. But this is a blocker as far as I understand so getting it to work is the top prio. Tests can come later.

**Comment by siemiatj** [comment]
> # 🔍 Dev test locally
> ## Status: ❌ Fail
> #### 10. Make sure there is no Quality Return Warehouse in the system
> 
> 10 Make sure there is no Quality Return Warehouse in the system
> #### 20. goto Shipment window; run action Select HU for Return; clear the active filter
> 
> 20 goto Shipment window; run action Select HU for Return; clear the active filter
> #### 30. Run action Return from customer
> 
> 30 Run action Return from customer
> #### 40. Error is shown, as expected
> 
> 40 Error is shown, as expected
> #### 50. Things get weird here
> 
>     * ✅ I can still see the rows
> 
>     * ❌ Action is now unavailable
> 
>     * ❌ Clicking the rows (first then second) repeatedly shows all sorts of errors in network tab
>       Shouldn't we have a new view here?   Idk who communicates that, or what should happen.
> 
> 
> 50 I can still see the rows - OK; Action is now unavailable - NOK; Clicking the rows (first then second) repeatedly shows all sorts of errors in network tab - NOK Shouldn't we have a new view here

This looks like a backend thing: https://recordit.co/AitSMBf94f . Deploying to a new dev instance.

File: frontend/src/reducers/viewHandler.js [hunk 2]
```diff
@@ -394,7 +416,8 @@ export default function viewHandler(state = initialState, action) {
     }
 
     case DELETE_VIEW: {
-      const id = action.payload.id;
+      const { id, isModal } = action.payload;
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply frontend/src/reducers/viewHandler.js:421]: Adding this once would really suffice.

**Comment by TheBestPessimist** [comment]
# 🔍 dev test locally

## Status: :question: Have questions

Performed on localhost


- :white_check_mark: Rows sent by backend are displayed correctly in frontend - ok afais:
	image


There still is the bug where backend filtering is broken after running an action which fails. This was shown in the previous dev test: https://github.com/metasfresh/metasfresh/pull/7020#pullrequestreview-449556983
This bug will be fixed by @teosarca in https://github.com/metasfresh/mf15/issues/84 and it is out of the scope of this task.


@siemiatj last thing needed is to go over the changes requested by @petrican in his first review. After those are done, i will dev test once more before merging.

**Comment by siemiatj** [comment]
@TheBestPessimist I don't really see a reason to focus on changing how parameters are sent to 20 functions instead of adding more test coverage or working on other bugs. So I won't fix it in this PR.

**Comment by TheBestPessimist** [comment]
@siemiatj then you can merge this as is.

**Pull request merged**